examples/optsCommands.c

```diff
@@ -0,0 +1,161 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ */
+
+#include <kdb.h>
+#include <kdbease.h>
+#include <kdbhelper.h>
+#include <kdbopts.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+extern char ** environ;
+
+#define BASE_KEY "/sw/org/kdbdummy/#0/current"
+#define SPEC_BASE_KEY "spec" BASE_KEY
+
+static KeySet * createSpec (void)
+{
+	return ksNew (10, keyNew (SPEC_BASE_KEY, KEY_META, "command", "", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/printversion", KEY_META, "description",
+			      "print version information and exit (ignoring all other options/commands/parameters)", KEY_META, "opt", "v",
+			      KEY_META, "opt/arg", "none", KEY_META, "opt/long", "version", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/getter", KEY_META, "description", "get a key's value", KEY_META, "command", "get", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/getter/verbose", KEY_META, "description",
+			      "print additional information about where the value comes from", KEY_META, "opt", "v", KEY_META, "opt/long",
+			      "verbose", KEY_META, "opt/arg", "none", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/getter/keyname", KEY_META, "description", "name of the key to read", KEY_META, "args",
+			      "indexed", KEY_META, "args/index", "0", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter", KEY_META, "description", "set a key's value", KEY_META, "command", "set", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter/verbose", KEY_META, "description",
+			      "print additional information about where the value will be stored", KEY_META, "opt", "v", KEY_META,
+			      "opt/long", "verbose", KEY_META, "opt/arg", "none", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter/keyname", KEY_META, "description", "name of the key to write", KEY_META, "args",
+			      "indexed", KEY_META, "args/index", "0", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/setter/value", KEY_META, "description", "value to be written", KEY_META, "args", "indexed",
+			      KEY_META, "args/index", "1", KEY_END),
+		      keyNew (SPEC_BASE_KEY "/dynamic/#", KEY_META, "description", "dynamically call a user-supplied command", KEY_META,
+			      "args", "remaining", KEY_END),
+		      KS_END);
+}
+
+int main (int argc, const char ** argv)
+{
+	KeySet * ks = createSpec ();
+	Key * errorKey = keyNew (BASE_KEY, KEY_END);
+
+	int result = elektraGetOpts (ks, argc, argv, (const char **) environ, errorKey);
+	if (result == -1)
+	{
+		fprintf (stderr, "ERROR: %s\n", keyString (keyGetMeta (errorKey, "error/reason")));
+		keyDel (errorKey);
+		ksDel (ks);
+		return EXIT_FAILURE;
+	}
+
+	if (result == 1)
+	{
+		char * help = elektraGetOptsHelpMessage (errorKey, NULL, NULL);
```

Comment:
As examples can have more verbose comments, I would write here that --help will make this code execute.